[APM][OTel] Fix layers issue for ESQL visualization by adding `textBased` layers

jennypavlova opened the pull request:
Closes #215675  

## Summary

This PR fixes the issue with the  ESQL visualization by supporting `textBased` layers. For testing, there is a link to the  serverless env of this PR  and it should be tested EDOT Java SDK as shown in [the screenshot in this comment](https://github.com/elastic/kibana/pull/216358#pullrequestreview-2728718952)


<!--ONMERGE {"backportTargets":["8.16","8.17","8.18","8.x","9.0"]} ONMERGE-->

jennypavlova committed:
[APM][OTel] Fix layers issue for ESQL visualizarion by adding `textBased` layers

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

jennypavlova committed:
Merge branch 'main' into 215675-apm-apm-services-metrics-tab-doesnt-work-if-embedded-dashboard-includes-esql-visualizations

JonasKunz commented on x-pack/solutions/observability/plugins/apm/public/components/app/metrics/static_dashboard/dashboards/dashboard_catalog.ts:
```diff
@@ -14,6 +14,7 @@ export const existingDashboardFileNames = new Set([
   'classic_apm-otel_other-dotnet',
   'classic_apm-edot-nodejs',
   'classic_apm-edot-java',
+  'otel_native-otel_other-java',
```
```suggestion
  'otel_native-edot-java',
```

JonasKunz reviewed (commented):
It works ... kind of. The dashboard is displayed, but it seems to be in "edit mode":
There are dashed lines around the panels, the panels can be dragged around and it is possible to open the visualization settings via the pen-icon:
![image](https://github.com/user-attachments/assets/8435cbb3-a30e-403d-a5f0-4613baedde15)

jennypavlova committed:
Update x-pack/solutions/observability/plugins/apm/public/components/app/metrics/static_dashboard/dashboards/dashboard_catalog.ts

Co-authored-by: Jonas Kunz <j+github@kunzj.de>

jennypavlova commented:
> It works ... kind of. The dashboard is displayed, but it seems to be in "edit mode":
There are dashed lines around the panels, the panels can be dragged around and it is possible to open the visualization settings via the pen-icon:

@JonasKunz Thank you for checking! 
I would like to ask the @elastic/kibana-presentation for some help: Do you know why that could happen ( we are loading a static dashboard file which uses ES|QL visualization and it appearch to be in "edit mode" )? - in this draft we added the file and changed [the logic we have to parse the pannels](https://github.com/elastic/kibana/pull/216358/files#diff-fe2f73b2a32c4514dec706789c0f9178b67b36d3b898c6474d14b56f1adf65afL60) which is  then used it in the`getCreationOptions` [here](https://github.com/elastic/kibana/blob/63575a8320a7e021df79727ad66ed4cf01dd1e5f/x-pack/solutions/observability/plugins/apm/public/components/app/metrics/static_dashboard/index.tsx#L61-L93) (we pass `        viewMode: 'view'` btw)  which is then passed to the `DashboardRenderer`

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/288997)
* Commit: 7329325691ee4858a20430ae2d4414a4f38c2a50

### Failed CI Steps
* [Jest Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/288997#0195f16d-1f27-4e4f-b785-985453823e83)
* [Jest Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/288997#0195f192-db54-4b69-b0b6-a06dac434ad9)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/288997#0195f16d-1f29-4a38-a8b1-ca2c0a394476)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/288997#0195f16d-1f29-4a38-a8b1-ca2c0a394476) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/288997/jobs/0195f16d-1f29-4a38-a8b1-ca2c0a394476/artifacts/0195f18f-feda-41b8-9f06-28299c0365ab) Jest Tests #<span></span>11 / <TableTab /> should filter the table correctly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/288997#0195f16d-1f27-4e4f-b785-985453823e83) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/288997/jobs/0195f16d-1f27-4e4f-b785-985453823e83/artifacts/0195f192-c4c4-4540-8f09-9bb163860598) Jest Tests #<span></span>2 / to_esql should work with iana timezones that fall udner utc+0
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/288997#0195f192-db54-4b69-b0b6-a06dac434ad9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/288997/jobs/0195f192-db54-4b69-b0b6-a06dac434ad9/artifacts/0195f1a0-aa4b-470f-adfb-3bf54531d0b1) Jest Tests #<span></span>2 / to_esql should work with iana timezones that fall udner utc+0

cc @jennypavlova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288997","number":288997,"commit":"7329325691ee4858a20430ae2d4414a4f38c2a50"}],"number":288997}
buildkite-pr-comment-->

jennypavlova committed:
Merge branch 'main' into 215675-apm-apm-services-metrics-tab-doesnt-work-if-embedded-dashboard-includes-esql-visualizations

jennypavlova commented:
/ci

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/289100)
* Commit: ffffc0a1ec98849dab344a7567ad50ffd302e465

### Failed CI Steps
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/289100#0195f1f2-3951-486b-bbb6-0484978b8a38)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/289100#0195f1f2-3951-486b-bbb6-0484978b8a38) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/289100/jobs/0195f1f2-3951-486b-bbb6-0484978b8a38/artifacts/0195f212-a7c6-4c7e-9f2d-5fee4705fb33) Jest Tests #<span></span>15 / <TableTab /> should filter the table correctly

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f578f401d464c800fe6772325a7ed5722f54f3e1) | [after](https://github.com/elastic/kibana/commit/ffffc0a1ec98849dab344a7567ad50ffd302e465) | diff |
| --- | --- | --- | --- |
| `apm` | 1888 | 1889 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f578f401d464c800fe6772325a7ed5722f54f3e1) | [after](https://github.com/elastic/kibana/commit/ffffc0a1ec98849dab344a7567ad50ffd302e465) | diff |
| --- | --- | --- | --- |
| `apm` | 2.5MB | 2.6MB | +62.7KB |

<details><summary>Unknown metric groups</summary>

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/f578f401d464c800fe6772325a7ed5722f54f3e1) | [after](https://github.com/elastic/kibana/commit/ffffc0a1ec98849dab344a7567ad50ffd302e465) | diff |
| --- | --- | --- | --- |
| `apm` | 70 | 71 | +1 |


</details>


### History
* :broken_heart: [Build #288997](https://buildkite.com/elastic/kibana-pull-request/builds/288997) failed 7329325691ee4858a20430ae2d4414a4f38c2a50

cc @jennypavlova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289100","number":289100,"commit":"ffffc0a1ec98849dab344a7567ad50ffd302e465"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288997","number":288997,"commit":"7329325691ee4858a20430ae2d4414a4f38c2a50"}],"number":289100}
buildkite-pr-comment-->

jennypavlova committed:
Merge branch 'main' into 215675-apm-apm-services-metrics-tab-doesnt-work-if-embedded-dashboard-includes-esql-visualizations

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/289379)
* Commit: 187ba980b035be984438cbac3d3651714bb374c0
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-216358-187ba980b035`

### History
* :yellow_heart: [Build #289100](https://buildkite.com/elastic/kibana-pull-request/builds/289100) was flaky ffffc0a1ec98849dab344a7567ad50ffd302e465
* :broken_heart: [Build #288997](https://buildkite.com/elastic/kibana-pull-request/builds/288997) failed 7329325691ee4858a20430ae2d4414a4f38c2a50

cc @jennypavlova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289379","number":289379,"commit":"187ba980b035be984438cbac3d3651714bb374c0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289100","number":289100,"commit":"ffffc0a1ec98849dab344a7567ad50ffd302e465"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288997","number":288997,"commit":"7329325691ee4858a20430ae2d4414a4f38c2a50"}],"number":289379}
buildkite-pr-comment-->

jennypavlova commented:
Hi @JonasKunz,
I investigated the issue, and I see the same in our oblt lite env (before this change) for different charts: 
![image](https://github.com/user-attachments/assets/eb7fced4-51ff-4998-bdbe-f0b8584cb5ed)
I deployed a serverless env from this PR and it's the same result for different dashboards: 
![image](https://github.com/user-attachments/assets/5099b14d-d0f0-41e1-b53d-019ff70c8e2f)
![image](https://github.com/user-attachments/assets/408538db-459c-4f12-b2c8-2b52e0e3fc02)

I will open this PR as it will fix this issue and the "edit issue" we should address separately ( I will open a separate issue for that to collaborate with @elastic/kibana-presentation team to fix it - I am not sure if it's a bug as it worked like that before this one 🤔 so I guess a regression of something else or a dashboards change )

jennypavlova committed:
Merge branch 'main' into 215675-apm-apm-services-metrics-tab-doesnt-work-if-embedded-dashboard-includes-esql-visualizations

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/290354)
* Commit: 2081b70388be38fdd476215c2be174fa171f9b4b
* [Cloud Deployment](https://kibana-pr-216358.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #289379](https://buildkite.com/elastic/kibana-pull-request/builds/289379) succeeded 187ba980b035be984438cbac3d3651714bb374c0
* :yellow_heart: [Build #289100](https://buildkite.com/elastic/kibana-pull-request/builds/289100) was flaky ffffc0a1ec98849dab344a7567ad50ffd302e465
* :broken_heart: [Build #288997](https://buildkite.com/elastic/kibana-pull-request/builds/288997) failed 7329325691ee4858a20430ae2d4414a4f38c2a50

cc @jennypavlova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290354","number":290354,"commit":"2081b70388be38fdd476215c2be174fa171f9b4b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289379","number":289379,"commit":"187ba980b035be984438cbac3d3651714bb374c0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289100","number":289100,"commit":"ffffc0a1ec98849dab344a7567ad50ffd302e465"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288997","number":288997,"commit":"7329325691ee4858a20430ae2d4414a4f38c2a50"}],"number":290354}
buildkite-pr-comment-->

hegerchr commented:
Can we please backport it to 8.x?

jennypavlova committed:
Merge branch 'main' into 215675-apm-apm-services-metrics-tab-doesnt-work-if-embedded-dashboard-includes-esql-visualizations

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/290666)
* Commit: 4ef4fd56392c85435b15a7ebf1f03199e4d7d5ec
* [Cloud Deployment](https://kibana-pr-216358.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #290354](https://buildkite.com/elastic/kibana-pull-request/builds/290354) succeeded 2081b70388be38fdd476215c2be174fa171f9b4b
* :green_heart: [Build #289379](https://buildkite.com/elastic/kibana-pull-request/builds/289379) succeeded 187ba980b035be984438cbac3d3651714bb374c0
* :yellow_heart: [Build #289100](https://buildkite.com/elastic/kibana-pull-request/builds/289100) was flaky ffffc0a1ec98849dab344a7567ad50ffd302e465
* :broken_heart: [Build #288997](https://buildkite.com/elastic/kibana-pull-request/builds/288997) failed 7329325691ee4858a20430ae2d4414a4f38c2a50

cc @jennypavlova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290666","number":290666,"commit":"4ef4fd56392c85435b15a7ebf1f03199e4d7d5ec"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290354","number":290354,"commit":"2081b70388be38fdd476215c2be174fa171f9b4b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289379","number":289379,"commit":"187ba980b035be984438cbac3d3651714bb374c0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289100","number":289100,"commit":"ffffc0a1ec98849dab344a7567ad50ffd302e465"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288997","number":288997,"commit":"7329325691ee4858a20430ae2d4414a4f38c2a50"}],"number":290666}
buildkite-pr-comment-->

rmyz reviewed (approved):
Tested and it works :rocket:

JonasKunz commented:
Also gave it another go with some EDOT Java data. Dashboard looks fine and correct

kibanamachine commented:
Starting backport for target branches: 8.16, 8.17, 8.18, 8.x, 9.0

https://github.com/elastic/kibana/actions/runs/14310306879

kibanamachine commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.16|Backport failed because of merge conflicts|
|❌|8.17|**Backport failed because of merge conflicts**<br><br>You might need to backport the following PRs to 8.17:<br> - [[ML] AIOps: Change Point Detection in Dashboards embeddable fix (#217178)](https://github.com/elastic/kibana/pull/217178)<br> - [Change reporting CODEOWNERS to response ops (#213560)](https://github.com/elastic/kibana/pull/213560)|
|✅|8.18|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/217321">](https://github.com/elastic/kibana/pull/217321)|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/217323">](https://github.com/elastic/kibana/pull/217323)|
|✅|9.0|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/217324">](https://github.com/elastic/kibana/pull/217324)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 216358
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

JonasKunz commented:
I think a backport to the next 8.x release (8.19) would be sufficient, having it already in 8.18.x would be a nice-to-have. We don't need support in previous versions, I'm even not sure if everything works there correctly. I originally created the dashboard in 8.18, so I'm confident that it works there and looks correct.

mistic commented:
This PR didn't make it into the latest BC. Updating the labels.
